src/widget/wsearchlineedit.cpp

```diff
@@ -75,18 +79,24 @@ WSearchLineEdit::WSearchLineEdit(QWidget* pParent)
     // Set up a timer to search after a few hundred milliseconds timeout.  This
     // stops us from thrashing the database if you type really fast.
     m_debouncingTimer.setSingleShot(true);
-    connect(&m_debouncingTimer, SIGNAL(timeout()), this, SLOT(triggerSearch()));
-
-    connect(this, SIGNAL(textChanged(const QString&)), this, SLOT(updateText(const QString&)));
+    connect(&m_debouncingTimer,
```

Comment:
Please switch to the new type-safe connect() syntax

Reply:
Those are just format changes according to changes in library.cpp I noticed earlier.
I'm unfamiliar with the new syntax, could you please point me to some syntax documentation? the only related recent changes I found are those about QSignalMapper in https://github.com/mixxxdj/mixxx/pull/2427/files and from those I can't conclude how it's supposed to look like here.

Reply:
```diff
- connect(&m_debouncingTimer, SIGNAL(timeout()), this, SLOT(triggerSearch()));
+ connect(&m_debouncingTimer, &QTimer::timeout, this, &WSearchLineEdit::triggerSearch);
```

Reply:
ah okay. I didn't have trouble with this line but with `this, SIGNAL(textChanged(const QString&)) ...` at least I couldn't get rid of the eclipse errors

Reply:
I'd appreciate a learning ressources link anyway.

Reply:
https://doc.qt.io/qt-5/signalsandslots.html#signals-and-slots-with-default-arguments
Relevant part starts around the middle of the section

Reply:
thx. I read that beforehand, but was confused by the lambdas, and Eclipse is no help with the syntax.